osquery/tables/system/posix/shell_history.h

```diff
@@ -16,6 +16,10 @@
 namespace osquery {
 namespace tables {
 
+void genShellHistoryFromFile(const std::string& uid,
+                             const boost::filesystem::path& history_file,
+                             QueryData& results);
+
```

Comment:
Just a question, why do we need it in header?

Reply:
I am not sure that it is needed.  I was just trying to make the header file complete.

Reply:
The idea of header files is to expose the interface for other modules in the project. The guideline is simple, if you use something outside of module (in most cases .cpp file) or have intention to use it somewhere else - put it in header. Otherwise just keep it private. It permit you to keep interfaces clear and brief.
If one day you need to use something from inside of module you will expose it in header.
Does it make sense for you?

Reply:
Understood.  I’ll abandon.

Reply:
Thank you!

Reply:
Is this considered a refactor if I need it to add a test?  Should I submit this separately from the test?